### DISCUSSION TITLE
Loosely couple the topic and schemaId association (allow many to many association)

### DISCUSSION THREAD

**Discussion by shiv4289** [apache/pulsar#18851] [General]
Currently, schema in pulsar is tightly coupled with a topic (one-to-one association). As part of this enhancement, we want to make it a many-to-many association, This would allow for:

1. A topic to have multiple different versioned schemas.
2. One versioned schema list (schemaGroup?) could map to different topics.

To do this, we need to change the way schema and topic association works today. Schema (keyed by schemaId) should NOT be stored under a topic key. Schema and topic should be stored in different zookeeper directories. The association should be stored separately to allow many to many associations between topic and schema.

[Here](https://docs.google.com/document/d/1YZF7sOAF3EWKspKreDgPk1lTUqaFlIbj4PwKScWsjAc/edit#) is a proposed solution for the same

**Comment by addisonj** [comment]
@shiv4289 That document is not readable, did you not yet make it public?

**Comment by shiv4289** [comment]
> @shiv4289 That document is not readable, did you not yet make it public?

Done now. Thanks!
